File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
    function  solve(total, str, root){
        if (root==null){
            return
        }
        if (root.left==null && root.right==null){
            total.val = total.val + (Number(str)*10 + root.val)
        }
        // left ki call marega toh string mai jodunga
        solve(total, str+String(root.val), root.left)
        // right mai bi toh add karna hai
        solve(total, str+String(root.val), root.right)
    }
    const total={val:0}
    var str=""

    solve(total, str, root)

    return total.val

    
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    void solve(int &sum, TreeNode* root, int digit) {
        if (root == NULL) {
            return;
        }
        digit = digit * 10 + root->val;
        if (root->left == NULL && root->right == NULL) {
            sum += digit;
            return ;
            
        }
        // left ki call marunga
        solve(sum, root->left, digit);
        // right ki call marunga
        solve(sum, root->right, digit);
    }
    int sumNumbers(TreeNode* root) {
        int sum = 0;
        int digit = 0;

        solve(sum, root, digit);
        return sum;
    }
};